app/src/lib/stores/app-store.ts

```diff
@@ -1449,7 +1457,11 @@ export class AppStore {
       gitStore.loadContextualCommitMessage(),
       refreshSectionPromise,
       gitStore.loadUpstreamRemote(),
+      this._refreshPullRequests(repository),
```

Comment:
This'll end up getting called a ton, as we call `_refreshRepository` after a bunch of actions. It's our generic "refresh the git world" method. I'm not sure we need to refresh pull requests in most of those situations. And if we did we'd probably run up against the API rate limit pretty quick.

Reply:
I can add it one level up. The intent here was to limit the number of places you have to call this method. I think we had some bugs that were caused by forgetting to refresh the PRs at the right time.